test/e2e/config_reload_test.go

```diff
@@ -231,6 +231,164 @@ func TestConfigReloadOnClientAuthSecretChange(t *testing.T) {
 	}
 }
 
+// TestConfigReloadOnClientAuthSecretChange creates a secret containing authentication data for a NATS cluster.
+// This secret initially contains two users ("user-1" and "user-2") and the corresponding password.
+// Then, the test creates a NatsCluster resource that uses this secret for authentication, and makes sure that "user-1" can connect to the NATS cluster.
+// Finally, it removes the entry that corresponds to "user-1" from the authentication secret, and makes sure that "user-1" cannot connect to the NATS cluster anymore.
+func TestConfigReloadOnClientAuthFileChange(t *testing.T) {
+	// Skip the test if "ShareProcessNamespace" is not enabled.
+	f.Require(t, framework.ShareProcessNamespace)
+
+	var (
+		username1 = "user-1"
+		username2 = "user-2"
+		password1 = "pass-1"
+		password2 = "pass-2"
+		size      = 1
+		version   = "1.3.0"
+	)
+
+	var (
+		auth        natsconf.AuthorizationConfig
+		c           *nats.Conn
+		cas         *v1.Secret
+		d           []byte
+		natsCluster *natsv1alpha2.NatsCluster
+		err         error
+	)
+
+	// Create an object containing client authentication data for the NATS cluster.
+	auth = natsconf.AuthorizationConfig{
+		Users: []*natsconf.User{
+			{
+				User:     username1,
+				Password: password1,
+				Permissions: &natsconf.Permissions{
+					Publish: []string{
+						">",
+					},
+					Subscribe: []string{
+						">",
+					},
+				},
+			},
+			{
+				User:     username2,
+				Password: password2,
+				Permissions: &natsconf.Permissions{
+					Publish: []string{
+						">",
+					},
+					Subscribe: []string{
+						">",
+					},
+				},
+			},
+		},
+	}
+	// Serialize the object containing authentication data.
+	if d, err = json.Marshal(auth); err != nil {
+		t.Fatal(err)
+	}
+	// Create a secret containing authentication data.
+	if authfileSecret, err = f.CreateSecret(f.Namespace, "data", d); err != nil {
+		t.Fatal(err)
+	}
+	// Make sure we cleanup the secret after we're done testing.
+	defer func() {
+		if err = f.DeleteSecret(cas); err != nil {
+			t.Error(err)
+		}
+	}()
+
+	// Create a NatsCluster resource with a single member, having configuration reloading enabled and using the secret above for client authentication.
+	natsCluster, err = f.CreateCluster(f.Namespace, "test-nats-", size, version, func(natsCluster *natsv1alpha2.NatsCluster) {
+		natsCluster.Spec.Auth = &natsv1alpha2.AuthConfig{
```

Comment:
Only thing I changed in the test is this.    Do you want me to roll them together?

Also, how can I run these guys?

Reply:
There are some notes in here to run the tests https://github.com/nats-io/nats-operator/blob/master/docs/development/development.md#testing

You can also run a single one by doing something like:

```sh
cd $GOPATH/src/github.com/nats-io/nats-operator/test/e2e 
go test ./... -run TestConfigReloadOnNatsServiceRoleUpdates -v --kubeconfig=$HOME/.kube/config
```

Btw I'm also looking at doing a release later today so if it is ok with you I can try to finish the test and push a new images for the operator + reloader sidecar that include the changes.